core/src/main/java/com/linecorp/armeria/server/auth/Authorizer.java

```diff
@@ -37,4 +40,55 @@
      *     will not be authorized.
      */
     CompletionStage<Boolean> authorize(ServiceRequestContext ctx, T data);
+
+    /**
+     * Returns a new {@link Authorizer} that delegates the authorization request to the specified
+     * {@link Authorizer} if this {@link Authorizer} rejects the authorization request by returning
+     * a {@link CompletionStage} completed with {@code false}.
+     */
+    default Authorizer<T> orElse(Authorizer<T> nextAuthorizer) {
+        final Authorizer<T> self = this;
+        return new Authorizer<T>() {
+            @Nullable
+            private String strVal;
+
+            @Override
+            public CompletionStage<Boolean> authorize(ServiceRequestContext ctx, T data) {
+                return self.authorize(ctx, data).thenComposeAsync(result -> {
+                    if (result != null && result) {
```

Comment:
Our javadoc for `Authorizer` only mentions `true` or `false`, and I think an `Authorizer` that returns `null` should be considered invalid. Do we need this `null` check?

Reply:
Added some null checks and helpful exception messages.